tests/test_formatting.py

```diff
@@ -62,6 +63,24 @@ def test_index_slice(self):
         self.assertEqual(self._format(b"data[1 :1-1];").rstrip(), b"data[1 : 1 - 1];")
         self.assertEqual(self._format(b"data[f(): ];").rstrip(), b"data[f() :];")
 
+    def test_format_comment_separator(self):
+        """Validates that we preserve the separator before a comment. This is a
+        regression test for #62."""
+        code = textwrap.dedent(
+            """\
+               event zeek_init()
+               \t{ }
+               # Comment on next line.
+
+               global x = 42; # Comment on same line.
+            """
+        ).encode()
```

Comment:
Unless I'm doing something completely wrong, I can't reproduce #62 with the latest zeek-script release (or when running from the `main` branch). The test seems useful, but it also passes if I don't pick up the MinorCommentFormatter change.

```
$ docker run --entrypoint=/bin/bash --rm -it -v $(pwd):/src -w /src python:3
# pip install zeekscript
# cat nl-preserve.zeek 
event zeek_init()
        {}
# Comment on next line.

global x = 42; # Comment on same line.
# zeek-script format < nl-preserve.zeek 
event zeek_init()
        { }
# Comment on next line.

global x = 42; # Comment on same line.
```

Reply:
Looks like the previous version works if we drop the line with the `global` which I added later. I broke that out into a separate check now.

Reply:
Now I'm even more confused, but approving as I hope it makes sense to you and does obviously fix something you run into.

Reply:
I agree, this is super confusing and I also do not have a good handle on local reasoning here. At least we have a smallish test case now so this exact issue cannot happen anymore 😞.